implementation/test/VendingMachineTest.js

```diff
@@ -423,6 +423,15 @@ describe("VendingMachine", async function() {
           owner,
         ])
 
+        const success = await tbtcToken.approveAndCall.call(
+          redemptionScript.address,
+          depositValue.add(signerFee),
+          calldata,
+          {from: owner},
+        )
+
+        expect(success).to.equal(true)
+
```

Comment:
If we add this, do we still need the `approveAndCall` below?

Reply:
Yeah, this one is .call so we don't get any state change

Reply:
So that we can observe the success, I guess…

These feel like separate tests. Any reason we've glommed them on to the existing ones rather than firing up a separate test that handles `approveAndCall` specifically?